vulcanai/models/metrics.py

```diff
@@ -1012,3 +1014,85 @@ def cross_validate(network, data_loader, k, epochs,
             logger.info(all_results)
             return all_results
 
+    def conduct_sensitivity_analysis(self, network, data_loader, filename,
+                                     features, cutoff=20):
+        """
+        Will conduct tests to figure out directionality of features by finding all the unique feature values present in
+        the dataset and setting the feature for all examples to every unique value of that feature.
+
+        Parameters
+        ----------
+            network : BaseNetwork
+                Network descendant of BaseNetwork.
+            data_loader : DataLoader
+                A DataLoader object which contains all the data
+            filename: string
+                The name of the file to save the test results to
+            cutoff: int
+                Maximum number of unique feature values for a particular feature that will be tested
+
+        Returns
+        -------
+        #TODO
+        """
+
+        if network._num_classes is None or network._num_classes == 0:
+            raise ValueError('There\'s no classification layer')
+
+        dataset = data_loader.dataset
+
+        test_df = None
+        col_headers = None
+
+        for feature in range(len(features)):
+            feature_val_list = dataset[:][0][:, feature].tolist()
```

Comment:
so in this case the feature could simply be an index? that's a good solution. did you test it with both?

Reply:
I haven't tested it with both, but it should work just as well. I think I should actually make the default value of features None, such that if it is None, I just create a new list that goes from 1 to size of num columns of data_loader.dataset

Reply:
yeah I like that!